internal/services/fabric/fabric_capacity_resource_test.go

```diff
@@ -0,0 +1,212 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/fabric/2023-11-01/fabriccapacities"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type FabricFabricCapacityResource struct{}
+
+func TestAccFabricFabricCapacity_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_fabric_capacity", "test")
+	r := FabricFabricCapacityResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccFabricFabricCapacity_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_fabric_capacity", "test")
+	r := FabricFabricCapacityResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccFabricFabricCapacity_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_fabric_capacity", "test")
+	r := FabricFabricCapacityResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccFabricFabricCapacity_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_fabric_capacity", "test")
+	r := FabricFabricCapacityResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r FabricFabricCapacityResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := fabriccapacities.ParseCapacityID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.Fabric.FabricCapacitiesClient
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return pointer.To(false), nil
+		}
```

Comment:
In the majority of cases there is no need to check for 404 in the Exists function for tests so this is actually unnecessary. 

I've been seeing this regularly as of late @ms-zhenhua could you please make sure to flag these in any future PR reviews that you do?
```suggestion
```

Reply:
Sure, thank you for the reminder. 
BTW, according to the latest guidance, `resp, err := client.Get(ctx, *id)` should be replaced with `resp, err := clients.Fabric.FabricCapacitiesClient.Get(ctx, *id)`

Reply:
Fixed.

Reply:
That's right, thanks for pointing that out @ms-zhenhua